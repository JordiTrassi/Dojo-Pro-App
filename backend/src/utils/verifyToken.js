const jwt = require('jsonwebtoken');

function verifyToken(err, user, res) {
  if (err) {
    return res.sendStatus(403);
  }

  // eslint-disable-next-line no-underscore-dangle
  const data = { _id: user._id, email: user.email };

  const token = jwt.sign(
    { user: data },
    process.env.JWT_SECRET,
    { expiresIn: process.env.TOKEN_EXPIRES_IN },
  );

  return res.json({
    token,
  });
}

module.exports = {
  verifyToken,
};
