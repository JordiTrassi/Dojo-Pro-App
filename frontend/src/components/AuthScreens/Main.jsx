import React from 'react';
import {
  SafeAreaView,
  View,
  ImageBackground,
  Image,
  Text,
  TouchableOpacity,
} from 'react-native';
import PropTypes from 'prop-types';
import styles from './Main_Styles';

const image = {
  uri: 'https://www.robinju.com/wp-content/uploads/2012/10/Fushimi-Inari-011.jpg',
};
const titleimage = {
  uri: 'https://fontmeme.com/permalink/210902/8337e2e2aeed704e1256a194f328c52c.png',
};

export default function Main({ navigation }) {
  const onPressLogin = () => { navigation.navigate('LoginScreen'); };
  const onPressRegister = () => { navigation.navigate('RegisterScreen'); };
  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.imageStack}>
        <ImageBackground
          source={image}
          resizeMode="contain"
          style={styles.image}
        >
          <Image source={titleimage} style={styles.titleimage} />
        </ImageBackground>
        <TouchableOpacity
          testID="loginBtn"
          style={styles.button2}
          onPress={onPressLogin}
        >
          <Text style={styles.logIn}>Log In</Text>
        </TouchableOpacity>
        <TouchableOpacity
          testID="registerBtn"
          style={styles.button}
          onPress={onPressRegister}
        >
          <Text style={styles.register}>Register</Text>
        </TouchableOpacity>

      </View>
    </SafeAreaView>
  );
}

Main.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
};
