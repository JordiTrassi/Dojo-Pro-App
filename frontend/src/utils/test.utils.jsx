/* eslint-disable react/prop-types */
import React from 'react';
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import configureStore from '../redux/store';

const customRender = (ui, initialState, options) => {
  const AllTheProviders = ({ children }) => (
    <Provider store={configureStore(initialState)}>
      { children }
    </Provider>
  );

  return render(ui, {
    wrapper: AllTheProviders,
    ...options,
  });
};

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render };
