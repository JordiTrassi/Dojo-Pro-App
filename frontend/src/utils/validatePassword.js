export default function validatePassword(password) {
  const regex = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[a-zA-Z\d]{8,}$/;
  return regex.test(password);
}
