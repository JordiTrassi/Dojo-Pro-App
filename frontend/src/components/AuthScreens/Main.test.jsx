/* eslint-disable no-undef */
import React from 'react';
import { render, fireEvent } from '../../utils/test.utils';
import Main from './Main';

const navigation = {
  navigate: jest.fn(),
  push: jest.fn(),
};
describe('Given a Character Component', () => {
  let screen;
  beforeEach(() => {
    screen = render(<Main navigation={navigation} />);
  });
  describe('When it is rendered', () => {
    test('Then screen should match the snapshot', () => {
      expect(screen).toMatchSnapshot();
    });
  });
  describe('And you click on loginButton', () => {
    test('Then navigation.push should have been called', () => {
      const loginButton = screen.getByTestId('loginBtn');
      fireEvent.press(loginButton);
      expect(navigation.navigate).toHaveBeenCalled();
    });
  });
  describe('And you click on registerButton', () => {
    test('Then navigation.push should have been called', () => {
      const registerButton = screen.getByTestId('registerBtn');
      fireEvent.press(registerButton);
      expect(navigation.navigate).toHaveBeenCalled();
    });
  });
});
