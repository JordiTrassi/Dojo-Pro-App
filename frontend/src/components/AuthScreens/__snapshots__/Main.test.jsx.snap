// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Character Component When it is rendered Then screen should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "rgba(15,15, 15,0)",
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "height": 761,
        "marginLeft": -242,
        "width": 953,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "height": 819,
          "left": 0,
          "position": "absolute",
          "top": -100,
          "width": 933,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "uri": "https://www.robinju.com/wp-content/uploads/2012/10/Fushimi-Inari-011.jpg",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": 819,
              "width": 933,
            },
            undefined,
          ]
        }
      />
      <Image
        source={
          Object {
            "uri": "https://fontmeme.com/permalink/210902/8337e2e2aeed704e1256a194f328c52c.png",
          }
        }
        style={
          Object {
            "height": 50,
            "left": 265,
            "top": 240,
            "width": 350,
          }
        }
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#E6E6E6",
          "borderColor": "#000000",
          "borderRadius": 20,
          "borderWidth": 3,
          "elevation": 5,
          "height": 65,
          "left": 280,
          "opacity": 1,
          "position": "absolute",
          "shadowColor": "rgba(0,0,0,1)",
          "shadowOffset": Object {
            "height": 3,
            "width": 3,
          },
          "shadowOpacity": 0.35,
          "shadowRadius": 0,
          "top": 656,
          "width": 135,
        }
      }
      testID="loginBtn"
    >
      <Text
        style={
          Object {
            "color": "#121212",
            "fontFamily": "arial",
            "fontSize": 28,
            "marginTop": 13,
            "textAlign": "center",
          }
        }
      >
        Log In
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#E6E6E6",
          "borderColor": "#000000",
          "borderRadius": 20,
          "borderWidth": 3,
          "elevation": 5,
          "height": 65,
          "left": 461,
          "opacity": 1,
          "position": "absolute",
          "shadowColor": "rgba(0,0,0,1)",
          "shadowOffset": Object {
            "height": 3,
            "width": 3,
          },
          "shadowOpacity": 0.35,
          "shadowRadius": 0,
          "top": 656,
          "width": 135,
        }
      }
      testID="registerBtn"
    >
      <Text
        style={
          Object {
            "color": "#121212",
            "fontFamily": "arial",
            "fontSize": 28,
            "marginTop": 13,
            "textAlign": "center",
          }
        }
      >
        Register
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
